lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -246,23 +249,37 @@ def get(self, request, *args, **kwargs):
             access_expiration = get_access_expiration_data(request.user, course_overview)
             cert_data = get_cert_data(request.user, course, enrollment.mode) if is_enrolled else None
 
-            # Only show the set course goal message for enrolled, unverified
-            # users in a course that allows for verified statuses.
-            is_already_verified = CourseEnrollment.is_enrolled_as_verified(request.user, course_key)
-            if not is_already_verified and has_course_goal_permission(request, course_key_string,
-                                                                      {'is_enrolled': is_enrolled}):
-                course_goals = {
-                    'goal_options': valid_course_goals_ordered(include_unsure=True),
-                    'selected_goal': None
-                }
-
-                selected_goal = get_course_goal(request.user, course_key)
-                if selected_goal:
-                    course_goals['selected_goal'] = {
-                        'key': selected_goal.goal_key,
-                        'text': get_course_goal_text(selected_goal.goal_key),
+            if COURSE_GOALS_NUMBER_OF_DAYS_GOALS.is_enabled():
+                if (is_enrolled and ENABLE_COURSE_GOALS.is_enabled(course_key)):
+
+                    course_goals = {
+                        'selected_goal': None
+                    }
+
+                    selected_goal = get_course_goal(request.user, course_key)
+                    if selected_goal:
+                        course_goals['selected_goal'] = {
+                            'days_per_week': selected_goal.days_per_week,
+                            'subscribed_to_goal_reminders': selected_goal.subscribed_to_goal_reminders,
+                        }
```

Comment:
If you wanted to be extra fancy and Django-y, you could make a DRF serializer for a goal object and just have the serializer do this kind of stuff for you. Where you set the selected goal as context or something. But this is good, especially as a rollout. Just struck me, since this whole section is just manually serializing. Maybe something we could do in future if we're feeling fancy.

Reply:
i never have a desire to be fancy although happy to change this is you feel there is a great reason why that is better - as it is not as familiar with the pros/cons of doing things in the serializers